Update caching.md (#6340)
circleci/circleci-docs: jekyll/_cci2/caching.md

### Clearing cache
{: #clearing-cache }

> Caches currently cannot be cleared. If you need to generate a new set of caches because you have updated language or build management tool versions, simply update the cache key, similar to the previous example.

Reviewer: ```suggestion
Caches cannot be cleared. If you need to generate a new set of caches because you have updated language or build management tool versions, simply update the cache key, similar to the previous example.
```

> Caches currently cannot be cleared. If you need to generate a new set of caches because you have updated language or build management tool versions, simply update the cache key, similar to the previous example.

Reviewer: ```suggestion
Caches currently cannot be cleared. If you need to generate a new set of caches you can update the cache key, similar to the previous example. You may wish to do this if you have updated language or build management tool versions.
```

> Updating the cache key on save and restore steps in the 'config.yml' file will then generate new sets of caches from that point onwards. Please note that older commits using the previous keys may still generate & save cache for builds, so remember to rebase after the 'config.yml' changes when possible

Reviewer 2: Missing a full-stop at the end: `...changes when possible.`

Author: Thanks for the good catch. Fixed.

> Updating the cache key on save and restore steps in the 'config.yml' file will then generate new sets of caches from that point onwards. Please note that older commits using the previous keys may still generate & save cache for builds, so remember to rebase after the 'config.yml' changes when possible.

Reviewer: ```suggestion
Updating the cache key on save and restore steps in your '.circleci/config.yml' file will then generate new sets of caches from that point onwards. Please note that older commits using the previous keys may still generate and save cache, so it is recommended that you rebase after the 'config.yml' changes when possible.
```

> You may create a new cache while still keeping the "older" cache by incrementing the cache version; however, be aware that you will be creating an additional cache, which will be available for 15 days and will increase your storage. As a general best practice, you should review what is currently being cached and reduce its size as much as possible.

Reviewer: ```suggestion
If you create a new cache by incrementing the cache version, the "older" cache is still stored. It is important to be aware that you are creating an additional cache, which will be available for 15 days. This method will increase your storage usage. As a general best practice, you should review what is currently being cached and reduce your storage usage as much as possible.
```

### Clearing cache
{: #clearing-cache }

> Caches currently cannot be cleared; however, if you need to get clean caches when your language or dependency management tool versions change, use a naming strategy similar to the previous example. Then change the cache key names in your `config.yml` file and commit the change to clear the cache.

Reviewer 2: I would potentially reword this to: 
" Caches currently cannot be cleared. If you need to generate a new set of caches because you've updated language or build management tool versions, simply update the cache key, similar to the previous example. 

Updating the cache key on save and restore steps in the `config.yml` file will then generate new sets of caches from that point onwards. Please note that older commits using the previous keys may still generate & save cache for builds, so remember to rebase after the `config.yml` changes when possible."

> You may create a new cache while still still keeping the "older" cache by incrementing the cache version; however, be aware that you will be creating an additional cache, which will be available for 15 days and will increase your storage. As a general best practice, you should review what is currently being cached and reduce its size as much as possible.

Reviewer 2: Duplicate "still" on line 202
